sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1080,7 +1077,7 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
 
     def apply(plan: LogicalPlan)
         : LogicalPlan = plan.resolveOperatorsUpWithPruning(AlwaysProcess.fn, ruleId) {
-      case i @ InsertIntoStatement(table, _, _, _, _, _) if i.query.resolved =>
+      case i @ InsertIntoStatement(table, _, _, _, _, _) =>
```

Comment:
This change is needed. We want to resolve the table first, then the column "DEFAULT" in the query. This means we can't wait for the query to be resolved before resolving the table.

Reply:
If we remove the pattern guard in this code, some operations on the "i.query" will fail later on. I create https://github.com/apache/spark/pull/44326 to fix